File: backend/internal/dto/shared_view.go
package dto

type SharedSummaryResponse struct {
	TotalAssetsShared    float64           `json:"total_assets_shared"`
	FormattedTotalAssets string            `json:"formatted_total_assets"`
	TotalDebts           float64           `json:"total_debts"`
	FormattedTotalDebts  string            `json:"formatted_total_debts"`
	NetWorthShared       float64           `json:"net_worth_shared"`
	FormattedNetWorth    string            `json:"formatted_net_worth"`
	UpcomingBills        []UpcomingBillDto `json:"upcoming_bills"`
	ForecastEndMonth     MoneyValue        `json:"forecast_end_month"`
	OwnerName            string            `json:"owner_name"`
}

File: frontend/src/services/sharedView.ts
import api from '../utils/api';
import type { Asset } from './assets';
import type { Debt } from './debts';
import type { UpcomingBill, MoneyValue } from './dashboard';

export interface SharedSummaryResponse {
  total_assets_shared: number;
  formatted_total_assets: string;
  total_debts: number;
  formatted_total_debts: string;
  net_worth_shared: number;
  formatted_net_worth: string;
  upcoming_bills: UpcomingBill[];
  forecast_end_month: MoneyValue;
  owner_name: string;
}

export const sharedViewService = {
  async getSharedSummary(): Promise<SharedSummaryResponse> {
    const res = await api.get('/shared-view/summary');
    return res.data.data;
  },

  async getSharedAssets(): Promise<Asset[]> {
    const res = await api.get('/shared-view/assets');
    return res.data.data || [];
  },

  async getSharedDebts(): Promise<Debt[]> {
    const res = await api.get('/shared-view/debts');
    return res.data.data || [];
  },

  async getSharedBills(): Promise<UpcomingBill[]> {
    const res = await api.get('/shared-view/bills');
    return res.data.data || [];
  },
};
export default sharedViewService;
